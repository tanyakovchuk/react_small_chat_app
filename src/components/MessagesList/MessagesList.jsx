import React from 'react';
import './MessagesList.scss';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export const MessagesList = ({ messages }) => {
  const lastIndexOfInterlocutor = messages.map(message => message.user)
    .lastIndexOf('second');

  return (
    <div className="App__content">
      {messages.map((element, index) => (
        <div
          className={
            classNames('App__content-element',
              { 'App__content-element--right': element.user === 'first' })
          }
          key={element.id}
        >
          <img
            src="./images/1.jpg"
            alt="user_icon"
            className={
              classNames('App__content-img',
                { 'App__content-img--active':
                  index === lastIndexOfInterlocutor
                  && element.user === 'second' })
            }
          />
          <span>
            <p className="App__content-message">
              {element.message}
            </p>
            <p className="App__content-date">
              {element.date}
            </p>
          </span>
        </div>
      ))}
    </div>
  );
};

MessagesList.propTypes = {
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      user: PropTypes.string.isRequired,
      message: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired,
    }),
  ).isRequired,
};
