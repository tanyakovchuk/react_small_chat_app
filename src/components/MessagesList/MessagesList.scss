.App {
  &__content {
    &-img {
      visibility: hidden;
      width: 40px;
      height: 40px;
      border-radius: 50%;
      margin-right: 20px;

      &--active {
        visibility: visible;
      }
    }

    &-element {
      display: flex;
      align-items: center;
      padding: 15px 10px;

      &--right {
        justify-content: flex-end;
        background-color: rgba(0, 0, 255, 0.03);
      }
    }

    &-message {
      margin: 0;
    }

    &-date {
      font-size: 12px;
      color: gray;
      margin: 0;
    }
  }
}
