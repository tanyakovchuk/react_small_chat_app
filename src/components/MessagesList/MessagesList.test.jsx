import React from 'react';
import renderer from 'react-test-renderer';
import {MessagesList} from './MessagesList';

test('MessagesList render', () => {
  
  const testMessages = [
    {
      id: 1,
      user: 'first',
      message: 'Hey here!',
      date: '19:10:05',
    },
    {
      id: 2,
      user: 'second',
      message: 'Hi! How are you?',
      date: '19:16:05',
    },
    {
      id: 3,
      user: 'second',
      message: 'Nice to see you here!',
      date: '19:17:50',
    },
  ];
  
  const component = renderer.create(
    <MessagesList messages={testMessages} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
