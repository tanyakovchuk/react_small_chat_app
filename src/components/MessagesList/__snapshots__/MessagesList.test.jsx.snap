// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList render 1`] = `
<div
  className="App__content"
>
  <div
    className="App__content-element App__content-element--right"
  >
    <img
      alt="user_icon"
      className="App__content-img"
      src="./images/1.jpg"
    />
    <span>
      <p
        className="App__content-message"
      >
        Hey here!
      </p>
      <p
        className="App__content-date"
      >
        19:10:05
      </p>
    </span>
  </div>
  <div
    className="App__content-element"
  >
    <img
      alt="user_icon"
      className="App__content-img"
      src="./images/1.jpg"
    />
    <span>
      <p
        className="App__content-message"
      >
        Hi! How are you?
      </p>
      <p
        className="App__content-date"
      >
        19:16:05
      </p>
    </span>
  </div>
  <div
    className="App__content-element"
  >
    <img
      alt="user_icon"
      className="App__content-img App__content-img--active"
      src="./images/1.jpg"
    />
    <span>
      <p
        className="App__content-message"
      >
        Nice to see you here!
      </p>
      <p
        className="App__content-date"
      >
        19:17:50
      </p>
    </span>
  </div>
</div>
`;
